src/webgl/primitives.js

```diff
@@ -1203,4 +1203,323 @@ p5.RendererGL.prototype.line = function() {
+    }
+
+    var LUTLength = this._lookUpTableBezier.length;
+
+    if (arguments.length === 6) {
+      this.isBezier = true;
+
+      w_x = [
+        this.immediateMode._bezierVertex[0],
+        arguments[0],
+        arguments[2],
+        arguments[4]
+      ];
+      w_y = [
+        this.immediateMode._bezierVertex[1],
+        arguments[1],
+        arguments[3],
+        arguments[5]
+      ];
+
+      for (i = 0; i < LUTLength - 1; i++) {
+        _x =
+          w_x[0] * this._lookUpTableBezier[i][0] +
+          w_x[1] * this._lookUpTableBezier[i][1] +
+          w_x[2] * this._lookUpTableBezier[i][2] +
+          w_x[3] * this._lookUpTableBezier[i][3];
+        _y =
+          w_y[0] * this._lookUpTableBezier[i][0] +
+          w_y[1] * this._lookUpTableBezier[i][1] +
+          w_y[2] * this._lookUpTableBezier[i][2] +
+          w_y[3] * this._lookUpTableBezier[i][3];
+        this.vertex(_x, _y);
+      }
+      this.immediateMode._bezierVertex[0] = arguments[4];
+      this.immediateMode._bezierVertex[1] = arguments[5];
+    } else if (arguments.length === 9) {
+      this.isBezier = true;
+
+      w_x = [
+        this.immediateMode._bezierVertex[0],
+        arguments[0],
+        arguments[3],
+        arguments[6]
+      ];
+      w_y = [
+        this.immediateMode._bezierVertex[1],
+        arguments[1],
+        arguments[4],
+        arguments[7]
+      ];
+      w_z = [
+        this.immediateMode._bezierVertex[2],
+        arguments[2],
+        arguments[5],
+        arguments[8]
+      ];
+      for (i = 0; i < LUTLength - 1; i++) {
+        _x =
+          w_x[0] * this._lookUpTableBezier[i][0] +
+          w_x[1] * this._lookUpTableBezier[i][1] +
+          w_x[2] * this._lookUpTableBezier[i][2] +
+          w_x[3] * this._lookUpTableBezier[i][3];
+        _y =
+          w_y[0] * this._lookUpTableBezier[i][0] +
+          w_y[1] * this._lookUpTableBezier[i][1] +
+          w_y[2] * this._lookUpTableBezier[i][2] +
+          w_y[3] * this._lookUpTableBezier[i][3];
+        _z =
+          w_z[0] * this._lookUpTableBezier[i][0] +
+          w_z[1] * this._lookUpTableBezier[i][1] +
+          w_z[2] * this._lookUpTableBezier[i][2] +
+          w_z[3] * this._lookUpTableBezier[i][3];
+        this.vertex(_x, _y, _z);
+      }
+      this.immediateMode._bezierVertex[0] = arguments[6];
+      this.immediateMode._bezierVertex[1] = arguments[7];
+      this.immediateMode._bezierVertex[2] = arguments[8];
+    }
+  }
+};
+
+p5.RendererGL.prototype.quadraticVertex = function() {
+  if (this.immediateMode._quadraticVertex.length === 0) {
+    throw 'vertex() must be used once before calling quadraticVertex()';
+  } else {
+    var w_x = [];
+    var w_y = [];
+    var w_z = [];
+    var t, _x, _y, _z, i;
+
+    t = parseFloat(0.0);
+
+    if (
+      this._lookUpTableQuadratic.length === 0 ||
+      this._lookUpTableQuadratic[this._lookUpTableQuadratic.length - 1] !==
```

Comment:
will this ever be true?

Reply:
True only when the first time quadraticVertex() is called. Also True when curveDetail() for which the lookUpTable was calculated the last time, isn't equal to the curveDetail() set afterwards.

So, in calls like these,

quadraticVertex()           // True
quadraticVertex()           // False
quadraticVertex()           // False

But for something like this,

quadraticVertex()            // True
quadraticVertex()            // False
curveDetail(5)                 
quadraticVertex()            // True
quadraticVertex()            // False

Reply:
oh, i see now. i was confused because it looked to me like _lookUpTableQuadratic/_lookUpTableBezier were arrays of coefficients. now i see that not all the elements of the array are the same type.

Reply:
Just the last element of the array....to store the current curveDetail 😁. Oh....Why not insert it into the first element? 😲

Reply:
does the current curveDetail _need_ to go in the array? wouldn't it be easier just to use a separate scalar?

Reply:
Yes, that can be done. But then we need a separate scalar for the look up table of bezierVertex() as well. @mlarghydracept what's your call?...

Reply:
i think that just at a conceptual level it's preferable to use arrays for holding homogeneous sets of data and use objects for structuring data of mixed types. i could go into a long diatribe about type theory, JIT optimizations, off-by-one bugs, eventual use of `for...of`, etc... but really it just seems cleaner to use the constructs that the language provides.

Reply:
I agree with @Spongman here. Arrays should probably hold homogenous sets of data. I feel like this will be more clear for future devs.